pkgs/development/haskell-modules/configuration-nix.nix

```diff
@@ -730,4 +730,29 @@ self: super: builtins.intersectAttrs super {
 
   # Tests access homeless-shelter.
   hie-bios = dontCheck super.hie-bios;
+
+  aeson-gadt-th = disableCabalFlag (doJailbreak (super.aeson-gadt-th)) "build-readme";
+
+  neuron = overrideCabal (super.neuron.override { rib = super.rib_0_9; })
+    (drv: rec {
+
+      src = fetchTarball {
+        # Temporarily deliver source from maralorns cloud to test before next neuron release
+        url = "https://cloud.maralorn.de/s/SktzC25ZJccH84n/download";
+        name = "neuron-0.3.1.0.tar.gz";
+        sha256 = "0dcn4rh21yd1q3l40yfjsm6hcca621r8dh944cqvjh2bw6bxv2mi";
+      };
+      # neuron expects the neuron-search script to be in PATH at built-time.
+      executableHaskellDepends = drv.executableHaskellDepends or [ ] ++ [
+        (pkgs.runCommand "neuron-search" {
```

Comment:
@cdepillabout Do you think this makeWrapper solution is fine?

Reply:
I think this is basically on the right track.

I'd propose a few small nitpicks:

- Technically this shouldn't be `executableHaskellDepends`, but something like `executableToolDepends` or maybe `executableSystemDepends` (depending on whether it should go into `nativeBuildInputs` or just `buildInputs`).
- There are helper functions to build PATHs.  Grep for `makeBinPath` in the top-level `lib/` directory in nixpkgs.
- I'd pull out the whole `runCommand` part into a let-statement somewhere and give it a comment.

I guess it might be possible to do the whole wrapping step for `neuron-search` in like the `postUnpack` phase of the `neuron` build itself, instead of creating a separate derivation for it.

Reply:
1. Okay makes sense. I think executableSystemDepends is the semantically the best.
2. Yeah I wondered about that. Nice, will have a look.
3. How far out? The runCommand needs the source. That‘s why it‘s in the derivation. If I don‘t find a solution for 4 I will just bump it as far as the scope permits.
4. (Re: postUnpack) Yes, that would be much better. But I just couldn‘t get that to build. It‘s no problem to write a wrapper to put the script into $out/bin but `neuron` then needs to have that dir in the $PATH at compile time. Can you tell my how to do that?

Reply:
TIL: Setting PATH in preBuild does not work, but it works in preConfigure or postUnpack.

Reply:
I decided to use preConfigure instead of postUnpack, because there PWD is already the source and $out/bin is already created.
I think I have addressed all of this.

Reply:
I take back the $out/bin part. Had to create that myself.